### PR TITLE
Msstitch

### PR DESCRIPTION
**Pull request by glormph**
About time I PRed this. Has been reworked a number of times.

Msstitch is a tool for postprocessing of ID and quant data, and generates PSM, peptide and protein/gene tables. It does not fit everything right now, but is currently leaning on Percolator, MSGFplus TSV, OpenMS consensusXML and Hardklor/kronik output.

Allows for flexible modular pipelines since the tools are fairly small.

### CODE REVIEW

File: tools/msstitch/.shed.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+owner: galaxyp
+name: msstitch_test2
```
Comment (bgruening) [inline tools/msstitch/.shed.yml]: wrong name?

File: tools/msstitch/.shed.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+categories:
+- Proteomics
+description: MSStitch
```
Comment (bgruening) [inline tools/msstitch/.shed.yml]: This should probably be a suite-definition file?

File: tools/msstitch/create_peptide_table.xml [hunk 3]
```diff
@@ -0,0 +1,116 @@
+<tool id="create_peptide_table" name="Create peptide table" version="1.2">
+    <description>from PSMs</description>
+    <requirements>
+        <requirement type="package" version="2.5">msstitch</requirement>
+    </requirements>
+    <stdio>
```
Comment (bgruening) [inline tools/msstitch/create_peptide_table.xml]: This could be replaced by `detect_errors`

File: tools/msstitch/create_peptide_table.xml [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+			    <param name="isobpatternratio" type="text" label="Narrower pattern for finding isobaric quant columns in PSM table to generate ratios" help="Regex to find quant columns, e.g. ^tmt10plex, ^itraq8plex"/>
+                            <repeat name="setdenoms" title="Biological sets denominators">
+			            <param name="setpattern" type="text" label="Sample set name for denominator" help="If no different sample sets are used, leave blank" optional="true" />
+			            <param name="denompatterns" type="text" label="Patterns for denominator isobaric quant columns in PSM table" help="Multiple are separated with space, e.g. ^tmt10plex_126 ^tmt10plex_127N." />
+                           </repeat>
+			   <param name="accessioncol" type="integer" value="" label="Accession column number in PSM table containing the peptides to quantify" />
+			   <param name="minint" type="float" value="0.1" label="Minimum intensity of PSM channel to include" />
+			   <conditional name="normalize">
+				   <param name="type" type="select" label="Use isobaric data?">
```
Comment (bgruening) [inline tools/msstitch/create_peptide_table.xml]: The indentation does not look like 4spaces.

File: tools/msstitch/datatypes_conf.xml [hunk 5]
```diff
@@ -0,0 +1,6 @@
+<?xml version="1.0"?>
+<datatypes>
+  <registration>
+    <datatype extension="kr" type="galaxy.datatypes.tabular:Tabular" subclass="true" />
```
Comment (bgruening) [inline tools/msstitch/datatypes_conf.xml]: Can we get this into Galaxy?

File: tools/msstitch/msslookup_protspace.xml [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+		label="FASTA database" help="Contains the protein sequences 
+		that will be used to create the lookup table" />
+        <param name="minlength" type="integer" label="Minimum length of peptides that will be filtered" value="" />
+    </inputs>
+    <outputs>
+        <data format="sqlite" name="output"/>
+    </outputs>
+    <tests>
+	    <test>
+		    <param name="input" value="ensembl.fasta" />
```
Comment (bgruening) [inline tools/msstitch/msslookup_protspace.xml]: Indentation seems to be off everywhere.

File: tools/msstitch/msslookup_quant.xml [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+                    </param>
+		    <param name="precursor|rttol" value="6" />
+		    <param name="precursor|mztol" value="20" />
+		    <param name="precursor|mztoltype" value="ppm" />
+		   <output name="output" md5="b355765289ca125a1a97d0816bf007b5" />
+	    </test>
+    </tests>
+    <help>
+	    This stores quantification data from OpenMS Isobaric Analyzer tools in an SQLite lookup database
+	    for later use in PSM quantification.
```
Comment (bgruening) [inline tools/msstitch/msslookup_quant.xml]: what about citations?

File: tools/msstitch/datatypes_conf.xml (hunk 5 above)
Comment (glormph) [reply tools/msstitch/datatypes_conf.xml]: Thanks @bgruening ! I have to read up a bit about the suite file and also fix the error detection thing, but here are some indent fixes. Cant believe how much my `filetype plugin indent on` in `.vimrc` would mess things up.

**Comment by glormph** [comment]
Thanks @bgruening ! I have to read up a bit about the suite file and also fix the error detection thing, but here are some indent fixes. Cant believe how much my filetype plugin indent on in .vimrc would mess things up.

**Comment by bgruening** [comment]
@glormph please see this here as an example: https://github.com/galaxyproject/tools-iuc/blob/master/tools/datamash/.shed.yml

**Comment by glormph** [comment]
Ok. Though I am not sure what ```suite:
  name: suite_datamash
  description: datamash performs basic numeric, textual and statistical operations on input textual data files
  type: repository_suite_definition
``` does

**Comment by bgruening** [comment]
I will create a meta-package in the TS like this one: https://toolshed.g2.bx.psu.edu/view/iuc/suite_datamash/2d85df079b20

This will just link to all sub-repos.

**Comment by bgruening** [comment]
@glormph it looks like the tests needs some love. Most of them passes!

**Comment by glormph** [comment]
Shit I tried to rebase on master but the commits look sketchy there.

**Comment by bgruening** [comment]
How can I help?

**Comment by bgruening** [comment]
@glormph anything I can do?

**Comment by glormph** [comment]
I think I messed up the rebase (this PR was old) because I see commits for other tools in this PR :(. Will get back to this later this week I hope.

**Comment by bgruening** [comment]
ping @glormph :)

**Comment by glormph** [comment]
Tried reopening but travis hiccups on a lint issue of another tool. Tests work on my planemo box. My guess is that somethign went wrong rebasing and now Travis thinks it needs to test a lot more tools. One solution would be to just copy the tools into a single-commit new branch and PR that.

**Comment by bgruening** [comment]
>  copy the tools into a single-commit new branch and PR that.

👍 can you do this?

**Comment by glormph** [comment]
Will give it a try!

**Pull request closed without merging**